processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java

```diff
@@ -187,7 +187,7 @@ public AggInterval next()
                   groupToRowIndex(relativeGroupId(1))
               ),
               Interval.of(
-                  groupToRowIndex(relativeGroupId(-lowerOffset)),
+                  groupToRowIndex(relativeGroupId(lowerOffset < 0 ? lowerOffset : -lowerOffset)),
```

Comment:
this doesn't look right - this is effectively: `-Math.abs(lowerOffset)`
this part was already working with relative offsets ; so I think it should be simply `relativeGroupId(lowerOffset)` - didn't that worked?